src/graph_notebook/gremlin/client_provider/utils_tinkerpop.py

```diff
@@ -0,0 +1,32 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+"""
+
+# Backport from TinkerPop 3.5.0 pre-release
+# https://github.com/apache/tinkerpop/blob/master/gremlin-python/src/main/python/gremlin_python/structure/io/util.py
+# https://github.com/apache/tinkerpop/pull/1314/files
+# https://github.com/apache/tinkerpop/pull/1383/files
+
+
+class HashableDict(dict):
```

Comment:
Let's rename this file to reflect what we're patching, `utils*` names can be cryptic when looking through package structure. Maybe `hashable_dict_patch`?